### PR TITLE
format .sh

### PR DESCRIPTION
**Pull request by turadg**
spawned from https://github.com/Agoric/agoric-sdk/pull/9179#discussion_r1546879844

## Description

adopt https://www.npmjs.com/package/prettier-plugin-sh to get [shfmt](https://github.com/mvdan/sh) within Prettier and our existing CI checks.

Options to consider: https://www.npmjs.com/package/prettier-plugin-sh#parser-options
described in https://github.com/mvdan/sh/blob/v2.6.4/syntax/printer.go#L32

### CODE REVIEW

File: golang/cosmos/scripts/protocgen.sh [hunk 1]
```diff
@@ -3,12 +3,12 @@
 set -eo pipefail
 
 protoc_gen_gocosmos() {
-  if ! grep "github.com/gogo/protobuf => github.com/regen-network/protobuf" go.mod &>/dev/null ; then
+  if ! grep "github.com/gogo/protobuf => github.com/regen-network/protobuf" go.mod &> /dev/null; then
```
Comment (turadg) [inline golang/cosmos/scripts/protocgen.sh:6]: The file had previously been formatted by `shfmt` but the Prettier plugin has the opposite default for this option:
https://github.com/mvdan/sh/blob/v2.6.4/syntax/printer.go#L32

Any rationale for one vs the other?

File: packages/telemetry/src/index.js [hunk 2]
```diff
@@ -61,9 +61,8 @@ export const getResourceAttributes = ({
       SDK_REVISION;
   }
   if (!resourceAttributes[SemanticResourceAttributes.SERVICE_INSTANCE_ID]) {
-    resourceAttributes[
-      SemanticResourceAttributes.SERVICE_INSTANCE_ID
-    ] = `${Math.random()}`;
+    resourceAttributes[SemanticResourceAttributes.SERVICE_INSTANCE_ID] =
+      `${Math.random()}`;
```
Comment (dckc) [inline packages/telemetry/src/index.js:65]: :clap: fewer lines

**Review by dckc — COMMENTED** [review]
seems like a lot of churn

I'm ambivalent about this level of consistency in shell scripts

File: .prettierrc.json5 [hunk 3]
```diff
@@ -0,0 +1,29 @@
+{
+  arrowParens: 'avoid',
+  overrides: [
+    {
+      // opt into prettier-plugin-jsdoc
+      files: [
+        'packages/ERTP/**/*.{js,ts}',
+        'packages/inter-protocol/**/*.{js,ts}',
+        'packages/store/**/*.{js,ts}',
+        'packages/vats/**/*.{js,ts}',
```
Comment (erights) [inline .prettierrc.json5:10]: Why is this not everything or almost everything? Or is this just an incremental step toward covering almost everything?

File: golang/cosmos/scripts/protocgen.sh (hunk 1 above)
Comment (erights) [reply golang/cosmos/scripts/protocgen.sh:6]: FWIW, I like the space. That's what I would have written manually.

File: package.json [hunk 4]
```diff
@@ -43,29 +44,6 @@
...
   },
-  "prettier": {
-    "arrowParens": "avoid",
-    "overrides": [
-      {
-        "files": [
-          "packages/ERTP/**/*.{js,ts}",
-          "packages/inter-protocol/**/*.{js,ts}",
-          "packages/store/**/*.{js,ts}",
-          "packages/vats/**/*.{js,ts}"
```
Comment (erights) [inline package.json:54]: Ah. I see that we're preserving the choice. Just moving where it is expressed. For this PR then, nevermind. But I'm still curious: should this be everything?

File: packages/cosmic-swingset/scripts/chain-entry.sh [hunk 5]
```diff
@@ -2,21 +2,21 @@
 # chain-entry.sh - decide whether to run ibc test setup or our chain
 set -e
 DIR=$(dirname -- "${BASH_SOURCE[0]}")
-DIR=`cd "$DIR" && pwd`
+DIR=$(cd "$DIR" && pwd)
```
Comment (erights) [inline packages/cosmic-swingset/scripts/chain-entry.sh:5]: Did it make this change automatically?

File: packages/deployment/bigdipper/bigdipper.sh [hunk 6]
```diff
@@ -5,20 +5,20 @@
 trap 'kill $(jobs -p) 2>/dev/null' EXIT
 
 NETWORK_URL=${1-"https://testnet.agoric.com"}
-ncf=`curl -Ss "$NETWORK_URL/network-config"`
-cn=`echo "$ncf" | jq -r '.chainName'`
+ncf=$(curl -Ss "$NETWORK_URL/network-config")
```
Comment (erights) [inline packages/deployment/bigdipper/bigdipper.sh:8]: Apparently so. How confident are you that this change is correctness preserving? Semantically equivalent? Or semantically preferred for some reason?

**Review by erights — APPROVED** [review]
For something this automated, I only skimmed. But of all differences I happened to glance at and give a moment's though to
- most seemed to be improvements or just as good
- I didn't spot a single change I thought was awful.

So, LGTM!

**Comment by erights** [comment]
> seems like a lot of churn

If we have high confidence that the reformatting is semantics preserving, then it is only syntactic churn. The only downside I see is PRs in progress that modify any of these files. But they could apply the formatter to their fork first to bring it back into alignment.

> I'm ambivalent about this level of consistency in shell scripts

Why? Unless there are unpleasant surprises, I'm for as much automated reformatting / pretty printing as possible. The resulting layout regularity makes it much easier to read a bunch of code.

File: .prettierrc.json5 (hunk 3 above)
Comment (turadg) [reply .prettierrc.json5:10]: The jsdoc formatter had some drawbacks so it was opt-in among package leaders. Feel free to add your favorite packages if you like.

File: packages/cosmic-swingset/scripts/chain-entry.sh (hunk 5 above)
Comment (turadg) [reply packages/cosmic-swingset/scripts/chain-entry.sh:5]: yes. all the changes to `.sh` files in this PR were by the formatter

File: packages/deployment/bigdipper/bigdipper.sh (hunk 6 above)
Comment (turadg) [reply packages/deployment/bigdipper/bigdipper.sh:8]: Fully confident it preserves. One reason it's better is it [allows nesting](https://stackoverflow.com/questions/9449778/what-is-the-benefit-of-using-instead-of-backticks-in-shell-scripts).

File: golang/cosmos/scripts/protocgen.sh (hunk 1 above)
Comment (turadg) [reply golang/cosmos/scripts/protocgen.sh:6]: same. I got used to the `shfmt` CLI taking it out, but I'll get used to adding the flag to keep it :)

**Comment by turadg** [comment]
> seems like a lot of churn

Is your concern about the `style` commit in git blame? Once it has a master commit sha we could configure the repo to ignore it: https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view

**Comment by dckc** [comment]
> > I'm ambivalent about this level of consistency in shell scripts
> 
> Why? Unless there are unpleasant surprises, I'm for as much automated reformatting / pretty printing as possible. The resulting layout regularity makes it much easier to read a bunch of code.

I don't find the results of this change much easier to read. I'm not that interested in consistent indentation because I'd rather not use shell at all for anything with non-trivial indentation or control flow.

If we're not migrating complex shell code to js (or python, for that matter) and we're interested in as much automation as possible, then [shellcheck](https://www.shellcheck.net/) is pretty much the state-of-the-art. That tool makes a much bigger difference in reliability than formatters, IME.

**Pull request merged**